home-manager.nix

```diff
@@ -43,43 +43,132 @@ in
       let
         link = file:
           pkgs.runCommand
-            "${replaceStrings [ "/" "." " " ] [ "-" "" "" ] file}"
+            "${sanitizeName file}"
             { }
             "ln -s '${file}' $out";
 
-        mkLinkNameValuePair = persistentStoragePath: fileOrDir: {
+        mkLinkNameValuePair = persistentStoragePath: file: {
           name =
             if cfg.${persistentStoragePath}.removePrefixDirectory then
-              dirListToPath (tail (splitPath [ fileOrDir ]))
+              dirListToPath (tail (splitPath [ file ]))
             else
-              fileOrDir;
-          value = { source = link (concatPaths [ persistentStoragePath fileOrDir ]); };
+              file;
+          value = { source = link (concatPaths [ persistentStoragePath file ]); };
         };
 
         mkLinksToPersistentStorage = persistentStoragePath:
           listToAttrs (map
             (mkLinkNameValuePair persistentStoragePath)
-            (cfg.${persistentStoragePath}.files ++ cfg.${persistentStoragePath}.directories)
+            (cfg.${persistentStoragePath}.files)
           );
       in
       foldl' recursiveUpdate { } (map mkLinksToPersistentStorage persistentStoragePaths);
 
+    systemd.user.services =
+      let
+        mkBindMountService = persistentStoragePath: dir:
+          let
+            mountDir =
+              if cfg.${persistentStoragePath}.removePrefixDirectory then
+                dirListToPath (tail (splitPath [ dir ]))
+              else
+                dir;
+            targetDir = concatPaths [ persistentStoragePath dir ];
+            mountPoint = concatPaths [ config.home.homeDirectory mountDir ];
+            name = "bindMount-${sanitizeName targetDir}";
+            script = pkgs.writeShellScript name ''
+              if ! mount | grep "${mountPoint}"; then
+                  ${pkgs.bindfs}/bin/bindfs -f --no-allow-other "${targetDir}" "${mountPoint}"
+              else
+                  echo "There is already an active mount at or below ${mountPoint}!" >&2
```

Comment:
Should this exit with an error? Otherwise, I'm envisioning a scenario where the `ExecStop` is executed on a mountpoint already "owned" by something else (and thus unmounting something one might not want it to).

Reply:
Ah, yes, you're right! I forgot add an `exit`.